File: diploma-service/src/main/java/org/arhor/diploma/service/AccountService.java
package org.arhor.diploma.service;

import org.arhor.diploma.service.dto.AccountDTO;
import org.springframework.security.core.userdetails.UserDetailsService;

import java.util.List;

public interface AccountService extends UserDetailsService {

  AccountDTO getAccountById(Long id);

  List<AccountDTO> getAccounts(int page, int size);
}

File: diploma-server/server-app/src/main/kotlin/org/arhor/diploma/service/AccountService.kt
package org.arhor.diploma.service

import kotlinx.coroutines.flow.Flow
import org.arhor.diploma.data.persistence.domain.Account
import org.arhor.diploma.service.dto.AccountDTO
//import org.springframework.security.core.userdetails.ReactiveUserDetailsService
//import org.springframework.security.core.userdetails.UserDetailsService

interface AccountService : CrudService<Account, AccountDTO, Long> /*, ReactiveUserDetailsService*/ {

    suspend fun getAccountById(id: Long): AccountDTO

    suspend fun getAccounts(page: Int, size: Int): Flow<AccountDTO>

    suspend fun createAccount(accountDTO: AccountDTO): AccountDTO

    suspend fun deleteAccount(id: Long)
}
